beets/ui/commands.py

```diff
@@ -1228,31 +1228,57 @@ def remove_items(lib, query, album, delete, force):
     """
     # Get the matching items.
     items, albums = _do_query(lib, query, album)
+    objs = albums if album else items
 
     # Confirm file removal if not forcing removal.
     if not force:
         # Prepare confirmation with user.
-        print_()
+        album_str = u" in {} album{}".format(
+                 len(albums), u's' if len(albums) > 1 else u''
+            ) if album else ""
+
         if delete:
             fmt = u'$path - $title'
-            prompt = u'Really DELETE %i file%s (y/n)?' % \
-                     (len(items), 's' if len(items) > 1 else '')
+            album_fmt = u'$path - $album'
+            prompt = u'Really DELETE'
+            prompt_all = u'Really DELETE {} file{}{}'.format(
+                 len(items), u's' if len(items) > 1 else u'', album_str
+            )
         else:
             fmt = u''
-            prompt = u'Really remove %i item%s from the library (y/n)?' % \
-                     (len(items), 's' if len(items) > 1 else '')
+            album_fmt = u''
+            prompt = u'Really remove from the library?'
+            prompt_all = u'Really remove {} item{}{} from the library?'.format(
+                 len(items), u's' if len(items) > 1 else u'', album_str
+            )
+
+        # Helpers for printing affected items
+        def fmt_track(t):
+            ui.print_(format(t, fmt))
+
+        def fmt_album(a):
+            ui.print_()
+            ui.print_(format(a, album_fmt))
+            ui.print_(u"which contains the tracks")
+            for i in a.items():
+                fmt_track(i)
```

Comment:
This display of albums and their contents is a little nonstandard for beets. What do you think about leaving the old behavior (just list the items)? This could still work by selecting albums when doing the interactive selection, but you would also still get the full list of files that would be deleted when "previewing" your selection.

Reply:
Ok, I agree that this is not really the current beets way. While I would really prefer some more structured output for albums, I shouldn't custom-build that into the remove command. Rather, the ui module should gain some unified facility to display albums. I don't have the time to revive the UI refactor, though, so I'll just restore the old behaviour :)

Reply:
Yeah, that sounds cool—I like the idea of having an "orthogonal" option to display items grouped by album.